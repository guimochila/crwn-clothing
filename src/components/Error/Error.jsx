import React from 'react';

import {
  ErrorImageOverlay,
  ErrorImageContainer,
  ErrorImageText,
} from './Error.styles';

class Error extends React.Component {
  constructor() {
    super();

    this.state = {
      hasError: false,
    };
  }

  static getDerivedStateFromError(error) {
    // Update state so the next render will show the fallback UI.
    return { hasError: true };
  }

  componentDidCatch(error, info) {
    // You can also log the error to an error reporting service
    console.log(error, info);
  }

  render() {
    if (this.state.hasError) {
      return (
        <ErrorImageOverlay>
          <ErrorImageContainer imageUrl="https://i.imgur.com/yW2W9SC.png" />
          <ErrorImageText>
            Ooops! Something went wrong, this page seems to be broken.
          </ErrorImageText>
        </ErrorImageOverlay>
      );
    }

    return this.props.children;
  }
}

export default Error;
