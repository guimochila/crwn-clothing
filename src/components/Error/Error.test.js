import React from 'react';
import { shallow, mount } from 'enzyme';

import Error from './Error';
import { ErrorImageContainer } from './Error.styles';

function ErrorChild() {
  return <div>Error</div>;
}

describe('Error boundary component', () => {
  let wrapper;
  const imageUrl = 'https://i.imgur.com/yW2W9SC.png';

  beforeEach(() => {
    wrapper = mount(
      <Error>
        <ErrorChild />
      </Error>,
    );
    wrapper.setState({ hasError: true });
  });
  test('displays error message when on error generated by child', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('ErrorImageContainer should render imageUrl prop', () => {
    expect(wrapper.find(ErrorImageContainer).prop('imageUrl')).toBe(imageUrl);
    const renderedErrorImageContainer = mount(
      <ErrorImageContainer imageUrl={imageUrl} />,
    );
    expect(renderedErrorImageContainer).toHaveStyleRule(
      'background-image',
      `url(${imageUrl})`,
    );
  });
});
