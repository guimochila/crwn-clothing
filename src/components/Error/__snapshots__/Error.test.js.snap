// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error boundary component displays error message when on error generated by child 1`] = `
.c0 {
  height: 60vh;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: inline-block;
  background-image: url(https://i.imgur.com/yW2W9SC.png);
  background-size: cover;
  background-position: center;
  width: 40vh;
  height: 40vh;
}

.c2 {
  font-size: 28px;
  color: #2f8e89;
}

<Error>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  height: 60vh;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <ErrorImageContainer
          imageUrl="https://i.imgur.com/yW2W9SC.png"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  display: inline-block;
  background-image: ",
                    [Function],
                    ";
  background-size: cover;
  background-position: center;
  width: 40vh;
  height: 40vh;
",
                  ],
                },
                "displayName": "ErrorImageContainer",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            imageUrl="https://i.imgur.com/yW2W9SC.png"
          >
            <div
              className="c1"
            />
          </StyledComponent>
        </ErrorImageContainer>
        <styled.h2>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": true,
                  "lastClassName": "c2",
                  "rules": Array [
                    "
  font-size: 28px;
  color: #2f8e89;
",
                  ],
                },
                "displayName": "styled.h2",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "h2",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <h2
              className="c2"
            >
              Ooops! Something went wrong, this page seems to be broken.
            </h2>
          </StyledComponent>
        </styled.h2>
      </div>
    </StyledComponent>
  </styled.div>
</Error>
`;
